playbooks/acs.yml

```diff
@@ -38,11 +38,14 @@
 - name: Database Role
   hosts: database
   gather_facts: false
+  vars:
+    test_env: "{{ repo_db_url == '' or (groups.syncservice and sync_db_url == '') }}"
   roles:
     - role: "../roles/postgres"
-      when: repo_db_url == "" or sync_db_url == ""
+      when: test_env
   post_tasks:
     - name: Make sure PostgreSQL is running
+      when: test_env
       ansible.builtin.service:
         name: "{{ postgresql_service }}"
         state: started
```

Comment:
we could also say that if you have external database then you should not have any host in the database group

a better fix would be to adopt the same external_something approach we are using for other components so become more natural having to deal with inventory to define the infrastructure, but for now looks fine to me telling to drop the host from the database group

Reply:
Couldn't agree more on aligning with other external services approach, though that's an heavier refactoring we can only plan for a major bump.
I have created OPSEXP-2408 for that (I was sure we already had a ticket for that but couldn't find it).
Are you suggesting we should not fix it and simply advertise one should make sure no database host remain in the inventoryif the repo_db_url are specified?

Reply:
yes we should not fix it, we are adding additional conditional logic here which is already implemented by `hosts` and the desired behaviour would be obtained by not defining the host in the database group, so for me it's more a misconfiguration than an actual bug.

Reply:
⬆️